src/sprite_timer.cpp

```diff
@@ -118,16 +128,6 @@ void Sprite_Timer::Update() {
 	digits[3].x = 32 + 8 * secs_10;
 	digits[4].x = 32 + 8 * secs_1;
 
-	if (Game_Temp::battle_running) {
-		SetY(SCREEN_TARGET_HEIGHT / 3 * 2 - 20);
-	}
-	else if (Game_Message::visible && Game_Message::GetRealPosition() == 0) {
-		SetY(SCREEN_TARGET_HEIGHT - 20);
-	}
-	else {
-		SetY(4);
-	}
-
 	SetZ(Priority_Timer);
```

Comment:
I don't like the solution but great that you found this bug. This affects everything when returning from battle, e.g. "Change Vehicle Location" in a battle will show the vehicle at the wrong position during fade in.
The correct way to fix this is the same one that solved the same issue for Teleports:

Add ``spriteset->Update();`` to the end of Continue().